Update continuous-integration.md

ccjjmartin opened the pull request:
SimpleTests are not supported on Pantheon.  I have an open ticket with Pantheon support discussing this: https://dashboard.pantheon.io/sites/025e433e-4745-4e86-bd6a-9fb252581f65#support/ticket/119011

Closes #

## Effect
PR includes the following changes:
-
-
-

## Remaining Work
- [ ] List any outstanding todos
- [ ] If needed

## Post Launch
To be completed by the docs team upon merge: 
- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] Update Status Report
- [ ] Archive from **Done** in Waffle

sarahg commented:
Support worked with Chris on this ticket and found a way to get this working using Terminus/Drush and the `exec php` command:

`terminus drush <site>.<env> -- exec php /path/to/your/test/script.sh --list`

We should update the doc page to mention this method.

ccjjmartin commented:
@sarahg I haven't forgot about this, the client that requested this task asked we work on it in Dec.  So a few more days and I should be able to swing around to this again.

ccjjmartin commented:
@sarahg This appears to be as far as we can get via the command line:

terminus drush SITE_NAME.SITE_ENV -- exec php `terminus drush SITE_NAME.SITE_ENV -- eval "echo DRUPAL_ROOT"`/scripts/run-tests.sh --url http://dev.SITE_NAME.com/ TEST_NAME

This errors out because it returns the html from the first test to my laptops shell instead of sending it back to simpletest (and I don't know enough about shell environments to be able to tackle this).  Without the `--url` option it calls localhost which makes the tests fail on Pantheon cli calls but running the tests from the UI works so Pantheon is capable of running simple tests (with the site_test module enabled) possibly not through the cli though.

Should we update the documentation to reflect as far as we got?

ccjjmartin commented:
It looks like I found the cause of our strange HTML dumps instead of successful passing test output.  It is the usual answer of "Have you tried clearing the cache?"  I get successful output but only after I have run a clear cache immediately before running the simple tests.  Here is my example CircleCI config that is working:

```
      - run:
          name: Test simpletest
          command: |
            if [ "${CIRCLE_BRANCH}" != "master" ]; then

              terminus drush $SITE_NAME.$ENV_NAME -- en site_test -y

              # If you don't clear the cache immediately before running tests
              # we get the html gibberish instead of a passing test.
              terminus drush $SITE_NAME.$ENV_NAME -- cc all

              # NOTE: It is a requirement to be running the latest version of
              # terminus in order to exclude the notice in shell output of the
              # embedded command to find the absolute path.
              terminus drush $SITE_NAME.$ENV_NAME -- exec php `terminus drush $SITE_NAME.$ENV_NAME -- eval "echo DRUPAL_ROOT" 2>/dev/null`/scripts/run-tests.sh --url http://$ENV_NAME-$SITE_NAME.pantheonsite.io Genomeweb

            fi
```

ccjjmartin commented:
@alexfornuto Updated the docs to reflect what we recently learned.  I am sure there will be some tone of voice changes / formatting changes etc to make it match the "Pantheon" voice but the technical details should be there.  Feel free to reach out if you have any questions.

alexfornuto commented:
@ccjjmartin is it correct to say that these instructions are only for D7, since the contrib module has no release for D8?

Please answer here, no need to update the documentation. I'm already making updates in a new branch and would like to avoid a merge conflict.

alexfornuto commented on source/_docs/continuous-integration.md:
````diff
@@ -19,15 +19,54 @@ You can use Terminus for scripting many operations. For example, a post-commit h
+
+To run tests the full command will look something like this:
+```
+terminus drush $SITE_NAME.$ENV_NAME -- exec php `terminus drush $SITE_NAME.$ENV_NAME -- eval "echo DRUPAL_ROOT" 2>/dev/null`/scripts/run-tests.sh --url http://$ENV_NAME-$SITE_NAME.pantheonsite.io Genomeweb
````
@ccjjmartin what is `Genomeweb` in this example?

ccjjmartin commented:
@alexfornuto Yes, the instructions are just for D7.  Looks like I updated the code in the comment above for that but forgot to update the code in the PR.  Please use the code from my comment above (https://github.com/pantheon-systems/documentation/pull/4238#issuecomment-448865918).  Genomeweb is the name of a simpletest test group, changed it to "OptionalTestGroup" in the comment.

alexfornuto commented:
@ccjjmartin please take a look at #4524, which continues the work started here.

ccjjmartin commented:
@alexfornuto Since there is a new PR that follows up on this one, if it is appropriate go ahead an close this one.

alexfornuto commented:
@ccjjmartin In this situation I'm in the habit of leaving it open so it will show as merged when the other PR is merged. That way the record shows your work as being added to the ~codebase~ docs without much digging.

alexfornuto commented:
@ccjjmartin unfortunately, because I rewrote git history a bit, this isn't showing as merged as I'd hoped. All the same, your commits are in the Docs code base forever. Thanks so much for contributing to the Pantheon Documentation! I've added you to our contributors page, as you've contributed a significant improvement to this doc.

ccjjmartin commented:
@alexfornuto No worries about the git history.  I appreciate you recognizing me by adding me to the contributors doc, it was a nice surprise.
